docs/DBT-dev.md

```diff
@@ -0,0 +1,112 @@
+# DBT development on `mattermost-analytics` project
+
+This document describes common development tasks on the new DBT project.
+
+`mattermost-analytics` was kickstarted as a project that follows best-practices defined in [DBT's documentation](https://docs.getdbt.com/guides/best-practices).
+The project follows a layered structure for the models:
+- **Staging layer**: Basic cleaning of source data.
+- **Intermediate layer**: Where reusable building blocks are introduced.
+- **Marts layer**: Purpose-specific models, to be used in Looker or other downstream integrations and tools.
+
+Model naming **MUST** follow the conventions defined in [DBT's documentation](https://docs.getdbt.com/guides/best-practices/how-we-structure/1-guide-overview).
+
+Column naming conventions **MUST** follow the [internal guidelines](https://mattermost.atlassian.net/wiki/spaces/DATAENG/pages/2422603777/DWH+Naming+Conventions). 
+
+## DBT packages
+
+The project is using DBT packages to make development faster and to ensure code quality.
+
+### DBT Utils ([git repo](https://github.com/dbt-labs/dbt-utils))
+
+DBT utils offers macros for easier common day-to-day operations and tests.
+
+### Codegen ([git repo](https://github.com/dbt-labs/dbt-codegen)) 
+
+Codegen offers macros & command line tools for generating DBT code. It is useful when incubating new data sources or adding
+documentation.
+
+### DBT Project Evaluator [git repo](https://github.com/dbt-labs/dbt-project-evaluator)
+
+DBT Project Evaluator checks the health of the project in terms of code quality. DBT project evaluator is running after 
+every hourly or nightly build. The results are visible in a Looker dashboard.
+
+## Development workflow
+
+1. Create a new branch. The convention is `<JIRA ticket>-<short description>`. For example `MM-123-add-telemetry-source `.
+2. Edit the project and implement any changes required (adding sources, adding and/or modifying models etc)
+3. Commit the changes and push the branch to github.
+4. Open a [Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). 
+5. As soon as the Pull Request (PR) is created, Cloud DBT will trigger a CI job that will run all new & modified models, as well as their downstream models. When the CI job finishes, all the models will be available in schema `dbt_cloud_pr_<account number>_<pr number>`.
+6. Validate the additions or changes and share the PR for review.
+7. As soon as there's enough approvers, the PR can be merged!
+
+## Common day-to-day operations
+
+This section describes common operations that might be required during DBT transformation development.
+
+> Note that most of the operations described in this section can also be executed in DBT's Cloud IDE. Please follow 
+> Codegen and Cloud DBT documentation. 
+
+### Adding a new source
```

Comment:
maybe add some info about (required/recommended) tests being added to some of these models.

Reply:
Added some ideas on the types of tests and how to implement them. It's hard to have a required/recommended part as each data source may have it's own peculiarities. For example  uniqueness constraints seems a good idea. Unfortunately there are examples such as CWS raw data including duplicate rows (due to stitch. In this specific case  it's easier to perform uniqueness tests on the staging layer rather than the source.